hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ConfigCheck.java

```diff
@@ -172,7 +184,13 @@ public int getId() {
     public void writeData(ObjectDataOutput out) throws IOException {
         out.writeUTF(groupName);
         //TODO @tkountis remove in 4.0
-        out.writeUTF(EMPTY_PWD);
+        if (clusterVersion.isLessThan(Versions.V3_9)) {
```

Comment:
I think this should be changed to `isUnknownOrLessThan` to explicitly indicate that the group password should be sent out when cluster version is not yet determined.